libs9s/s9srpcclient.cpp

```diff
@@ -4485,23 +4487,90 @@ S9sRpcClient::registerMongoDbCluster(
  * \param hosts the hosts that will be the member of the cluster (variant list
  *   with S9sNode elements).
  * \param osUserName the user name to be used to SSH to the host.
+ * \param version the version of the database to install.
+ * \param clusterType the type of the cluster to create. 
  * \returns true if the request sent and a return is received (even if the reply
  *   is an error message).
  *
  * This method will create a job that creates a redis sharded cluster
  */
 bool
-S9sRpcClient::createRedisSharded(
+S9sRpcClient::createRedisOrValkeySharded(
         const S9sVariantList &hosts,
         const S9sString      &osUserName,
-        const S9sString      &version)
+        const S9sString      &version,
+        const S9sString      &clusterType)
 {
     S9sOptions     *options = S9sOptions::instance();
     S9sVariantMap   request;
     S9sVariantMap   job = composeJob();
     S9sVariantMap   jobData = composeJobData();
     S9sVariantMap   jobSpec;
     S9sString       uri = "/v2/jobs/";
+    S9sString       jobClusterType;
+    S9sString       jobTitle;
+
+    if(clusterType.contains("redis"))
+    {
+        jobClusterType = "redis_sharded";
```

Comment:
I'd recommend to split this function to have separate functions for redis and valkey.